src/main/kotlin/io/kotlintest/TestBase.kt

```diff
@@ -30,19 +36,33 @@ abstract class TestBase : PropertyTesting(), Matchers, TableTesting {
    * @see config
    * @see TestBase.config
    */
-  open val defaultTestCaseConfig: TestConfig = TestConfig()
+  protected open val defaultTestCaseConfig: TestConfig = TestConfig()
 
-  // the root test suite which uses the simple name of the class as the name of the suite
-  // spec implementations will add their tests to this suite
-  internal val root = TestSuite(javaClass.simpleName, ArrayList<TestSuite>(), ArrayList<TestCase>())
+  /**
+   * Interceptors that intercepts the execution of the whole spec. Interceptors are executed from
+   * left to right.
+   */
+  protected open val specInterceptors: Iterable<(TestBase, () -> Unit) -> Unit> = listOf()
```

Comment:
Since it is `protected` and `open` and the type is `Iterable` I believe a subclass can assign a Set as a value and then the promise to execute from left to right is broken.